automation/src/test/java/org/wso2/carbon/testgrid/automation/executor/JMeterExecutorTest.java

```diff
@@ -0,0 +1,346 @@
+          expectedExceptionsMessageRegExp = TestGridConstants.JMETER_HOME + " environment variable not set.")
+    public void initTestJMeterHomeInvalid(String jMeterHome) throws TestAutomationException {
+        EnvironmentUtil.setEnvironmentVariable(TestGridConstants.JMETER_HOME, jMeterHome);
+        JMeterExecutor jMeterExecutor = new JMeterExecutor();
+        jMeterExecutor.init("", "");
+        EnvironmentUtil.unsetEnvironmentVariable(TestGridConstants.JMETER_HOME);
+    }
+
+    @Test(description = "Tests for valid JMETER_HOME paths")
+    public void initTestJMeterHomeValid() throws TestAutomationException {
+        EnvironmentUtil.setEnvironmentVariable(TestGridConstants.JMETER_HOME, "JMeter Home");
+        JMeterExecutor jMeterExecutor = new JMeterExecutor();
+        jMeterExecutor.init("", "");
+        EnvironmentUtil.unsetEnvironmentVariable(TestGridConstants.JMETER_HOME);
+    }
+
+    @Test(description = "Tests for valid JMETER_HOME not set",
+          expectedExceptions = TestAutomationException.class,
+          expectedExceptionsMessageRegExp = "JMETER_HOME environment variable not set.")
+    public void initTestJMeterHomeNotSet() throws TestAutomationException {
+        JMeterExecutor jMeterExecutor = new JMeterExecutor();
+        jMeterExecutor.init("", "");
+    }
+
+    @Parameters({"jmeterHome"})
+    @Test(description = "Test for executing JMeter")
+    public void testExecuteJMeter(String jmeterHome) throws URISyntaxException, TestAutomationException {
+        // Set JMeter home
+        EnvironmentUtil.setEnvironmentVariable(TestGridConstants.JMETER_HOME, jmeterHome);
+        // Set cloned test plan location.
+        ClassLoader classLoader = getClass().getClassLoader();
+        URL resource = classLoader.getResource("test-grid-is-resources");
+        Assert.assertNotNull(resource);
+
+        String testScript = Paths.get(resource.getPath(), "SolutionPattern22", "Tests", "JMeter", "solution22",
+                "src", "test", "jmeter", "Create-Test-Users.jmx").toAbsolutePath().toString();
+
+        // Create open port list
+        Port port = createPort(9443, "https");
+        List<Port> portList = Collections.singletonList(port);
+
+        // Creat host list
+        Host host = createHost("localhost", "wso2is-default", portList);
+        List<Host> hostList = Collections.singletonList(host);
+
+        Deployment deployment = createDeployment("single-node", hostList);
+
+        String testLocation = Paths.get(resource.getPath(), "SolutionPattern22", "Tests")
+                .toAbsolutePath().toString();
+        TestExecutor testExecutor = new JMeterExecutor();
+        testExecutor.init(testLocation, "solution22");
+        testExecutor.execute(testScript, deployment);
+
+        // Assert test result file
+        Assert.assertTrue(Files.exists(Paths.get(resource.getPath(), "SolutionPattern22", "Tests", "Results",
+                "Jmeter", "Create-Test-Users.jmx.xml")));
```

Comment:
Is it enough to only verify the existence of the file?

Reply:
No.With issue https://github.com/wso2-incubator/wso2-test-grid/issues/84 we are taking the results out programmatically via JMeter and will be handled there.